Source/Scene/WebMapTileServiceImageryProvider.js

```diff
@@ -228,13 +228,19 @@ function WebMapTileServiceImageryProvider(options) {
   var style = options.style;
   var tileMatrixSetID = options.tileMatrixSetID;
   var url = resource.url;
-  if (url.indexOf("{") >= 0) {
-    var templateValues = {
-      style: style,
-      Style: style,
-      TileMatrixSet: tileMatrixSetID,
-    };
-
+  var templateValues = {
+    style: style,
+    Style: style,
+    TileMatrixSet: tileMatrixSetID,
+  };
+
+  if (
+    (url.match(/{s}/g) || []).length === 1 &&
+    (url.match(/{/g) || []).length === 1
+  ) {
+    resource.setTemplateValues(templateValues);
```

Comment:
Why would we set the template values here? They are never used. Something like this would be simpler. We generally don't create empty arrays for no reason just to test the length.
```
    var bracketMatch = url.match(/{/g);
    if ( !defined(bracketMatch)  || (bracketMatch.length === 1 && /{s}/.test(url))) {
       // No brackets or 1 bracket and it belongs to {s}
        this._useKvp = true;
    } else {
       // A bracket not belonging to {s}
        resource.setTemplateValues(templateValues);
        this._useKvp = false;
    }
```

Reply:
@tfili Thank you for the feedback. I now see why it does not make sense to set the template values in this part of the code. I just added a commit that should address your suggestion.